### PR TITLE
Moved some of RideCreateAction checks into the Query member.

### PR DESCRIPTION
**Pull request by ZehMatt**

### CODE REVIEW

File: src/openrct2/actions/RideCreateAction.hpp [hunk 1]
```diff
@@ -310,5 +277,41 @@ struct RideCreateAction : public GameActionBase<GAME_COMMAND_CREATE_RIDE, GA_FLA
...
 
+private:
+    sint32 GetSubType() const
+    {
+        sint32 subType = rideSubType;
+
+        if (subType == RIDE_ENTRY_INDEX_NULL)
+        {
+            uint8 *availableRideEntries = get_ride_entry_indices_for_ride_type(rideType);
+            for (uint8 *rei = availableRideEntries; *rei != RIDE_ENTRY_INDEX_NULL; rei++)
```
Comment (duncanspumpkin) [inline src/openrct2/actions/RideCreateAction.hpp]: Use a better name than rei.

Comment (wolfreak99) [reply src/openrct2/actions/RideCreateAction.hpp]: i'd just use "i"

Comment (ZehMatt) [reply src/openrct2/actions/RideCreateAction.hpp]: Thats worse, its a pointer if you haven't noticed.

File: src/openrct2/ride/ride.c [hunk 2]
```diff
@@ -8353,3 +8353,39 @@ bool ride_entry_has_category(const rct_ride_entry * rideEntry, uint8 category)
...
+            }
+        }
+        if (subType == RIDE_ENTRY_INDEX_NULL)
+        {
+            subType = availableRideEntries[0];
+        }
+    }
+
+    return subType;
+}
```
Comment (janisozaur) [inline src/openrct2/ride/ride.c]: missing newline

File: src/openrct2/actions/RideCreateAction.hpp [hunk 3]
```diff
@@ -62,64 +63,41 @@ struct RideCreateAction : public GameActionBase<GAME_COMMAND_CREATE_RIDE, GA_FLA
...
-        if (rideIndex == -1) 
-        {
-            res->Error = GA_ERROR::DISALLOWED;
-            res->ErrorMessage = STR_TOO_MANY_RIDES;
-            return std::move(res);
-        }
 
         res->rideIndex = rideIndex;
-        res->rideColor = ride_get_random_colour_preset_index(rideType) | (ride_get_unused_preset_vehicle_colour(rideType, subType) << 8);
+        //res->rideColor = ride_get_random_colour_preset_index(rideType) | (ride_get_unused_preset_vehicle_colour(rideType, rideEntryIndex) << 8);
```
Comment (janisozaur) [inline src/openrct2/actions/RideCreateAction.hpp]: If there's no need for this code, just remove it, don't submit commented out code.

File: src/openrct2/actions/RideCreateAction.hpp (hunk 1 above)
Comment (wolfreak99) [reply src/openrct2/actions/RideCreateAction.hpp]: I mean, rei looks like it stands for Ride Entry Iterator, or Ride Entry Index. But I've always stuck with short variable names for iteration sequences just to tell if part of a code is part of some form of iteration or loop. maybe rideEntryIndex?

**Pull request merged**